### PR TITLE
feat(container): Allow use of podman container name instead of image name

### LINKED ISSUE #3592: The container indicator should use the container name
https://github.com/starship/starship/pull/3304 added support for a container indicator but, at least in the case of podman, it uses the value of "image" from `/run/.containerenv`. Instead, it should use the value of "name" if set, to avoid that different containers derived from the same image have the same indicator.

### PR DESCRIPTION
**Pull request by Fuzzy-Math**
#### Description

This PR adds configuration in the container module to allow displaying the container name instead of the container image. This is useful for when many containers share the same image (ie. toolbox containers). To enable, add the following to starship.toml
```
[container]
use_container_name = true
```

Closes #3592

#### How Has This Been Tested?

- [ ] I have tested using **MacOS**
- [X] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [X] I have updated the documentation accordingly.
- [X] I have updated the tests accordingly.

### CODE REVIEW

File: src/modules/container.rs [hunk 1]
```diff
@@ -37,16 +40,30 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
...
-                        .find_map(|l| {
-                            l.starts_with("image=\"").then(|| {
-                                let r = l.split_at(7).1;
-                                let name = r.rfind('/').map(|n| r.split_at(n + 1).1);
-                                String::from(name.unwrap_or(r).trim_end_matches('"'))
-                            })
+                .map(|buf| {
+                    buf.lines()
+                        .find_map(|line| {
+                            if config.use_container_name && line.starts_with("name") {
```
Comment (davidkna) [inline src/modules/container.rs:46]: Please change this to only iterate once over the lines.

File: src/modules/container.rs [hunk 2]
```diff
@@ -37,16 +40,30 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
...
+                                let (_, name) = line.split_once('=').unwrap_or(("", "podman"));
+                                Some(String::from(name.trim_matches('"')))
```
Comment (davidkna) [inline src/modules/container.rs:48]: Would `shellwords::split`work here?

File: src/configs/container.rs [hunk 3]
```diff
@@ -8,6 +8,7 @@ pub struct ContainerConfig<'a> {
     pub symbol: &'a str,
     pub style: &'a str,
     pub disabled: bool,
+    pub use_container_name: bool,
```
Comment (davidkna) [inline src/configs/container.rs:11]: Perhaps call this `prefer_container_name` instead, your code seems to include a fallback.

**Review by davidkna — CHANGES_REQUESTED** [review]

**Comment by spencergilbert** [comment]
I'd personally suggest the current behavior is a bug, or at least contradicting the documentation of the feature. I find the actual container name to be more useful to me than the name of the image, but it seemed like that is split between users - perhaps a configuration to toggle between the two is a good compromise.

**Comment by andytom** [comment]
Closing this PR as it has been superceded by a few others.

**Pull request closed without merging**